src/main/java/org/myrobotlab/io/FileIO.java

```diff
@@ -1424,5 +1426,35 @@ public static String normalize(String dirPath) {
       return dirPath.replace("\\", "/");
     }
   }
+  
+  public static boolean isExecutableAvailable(String command) {
+    try {
+        // Attempt to execute the command
+        Process process = java.lang.Runtime.getRuntime().exec(command);
+
+        // Check the exit value of the process
+        // If the process has terminated correctly, the command is available
+        if (process.waitFor() == 0) {
+            return true;
+        }
+
+        // Read any errors from the attempted command
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getErrorStream()))) {

```

Comment:
what happens if the stderr stream gets filled up and blocks before the process completes?  Seems like the waitFor above might cause a deadlock here...  (possibly?)

Reply:
isExecutableAvailable isn't expected to be a long running exe or block, the OS should return an error, or the exe should quickly return version, help or some other relevant information.  That at least was the intent.